api/envoy/extensions/network/dns_resolver/cares/v3/cares_dns_resolver.proto

```diff
@@ -47,4 +48,20 @@ message CaresDnsResolverConfig {
   // This option allows for number of UDP based DNS queries to be capped. Note, this
   // is only applicable to c-ares DNS resolver currently.
   google.protobuf.UInt32Value udp_max_queries = 5;
+
+  // The initial response timeout for DNS queries on the first attempt. This timeout value is used
+  // as the base for c-ares' exponential backoff algorithm on retries. Subsequent timeouts are
+  // scaled by factors of 2^(n-1) where n is the attempt number.
+  // See: https://github.com/c-ares/c-ares/blob/d73dd5181d7f8df36003225324cabcd551a6d9b4/src/lib/ares_query.c#L287
+  //
+  // Note: While c-ares defaults to 2 seconds, Envoy overrides this to 5 seconds to preserve the old behavior after
+  // getting some user reports of increased DNS resolution times.
+  google.protobuf.UInt64Value query_timeout_seconds = 6 [(validate.rules).uint64 = {gte: 1}];
+
+  // The maximum number of query attempts the resolver will make before giving up.
+  // Each attempt may use a different name server.
+  //
+  // Note: While c-ares defaults to 3 tries, Envoy overrides this to 4 tries to preserve the old behavior after
```

Comment:
The wording is unclear here.  I assume it means that Envoy defaults to 4, but that default can be overridden via this field.  But the text implies that Envoy may override what is set here, which doesn't make sense.

Reply:
@markdroth How about this?
_"Note: The c-ares library defaults to 3 attempts, but Envoy's default (if this field is unset) is 4 attempts. This adjustment was made to maintain the previous behavior after users reported an increase in DNS resolution times."_

Does this look better?

Reply:
That's much clearer, thank you!